### ISSUE TITLE
Update lesson to work with google sheets

### ISSUE THREAD

**Issue by colinquirk** [datacarpentry/spreadsheets-socialsci#97]
It's not clear to me if this lesson would work with google sheets out of the box, but considering there is a very small cost to getting google sheets working on your personal machine, I think it would be a nice alternative to libreoffice which has known issues reported in the lesson (e.g. different delimiters).

**Comment by angela-li** [comment]
Yes, agreed that Google Sheets may be a better alternative to LibreOffice. For data validation, Google Sheets also has more functionality.

**Comment by chris-prener** [comment]
I like this suggestion @colinquirk, and thanks for your patience! I agree with @angela-li that it would be perhaps preferred over LibreOffice.

However, I'm not in a place to do the makeover myself. Is this something you would like to contribute to, @colinquirk ?

**Comment by colinquirk** [comment]
Sorry, I don't think I'll be able to take this on. Maybe a new instructor will be interested in taking a shot.

**Comment by chris-prener** [comment]
No worries @colinquirk !

**Comment by kerchner** [comment]
Has anyone yet tried running through the material to at least determine at which points Google Sheets users would have to do something different?

**Comment by chris-prener** [comment]
not that I am aware of, @kerchner!

**Comment by dolsysmith** [comment]
Having taught this lesson last week with Excel, I just ran through the demonstrations and exercises using Google Sheets. With one exception -- in the Data Validation lesson -- the functionality in Sheets tracks closely to Excel for this content. See my summary below:

- Lesson 2, Formatting Data Tables in Spreadsheets
  - SAFI_messy.xslx opens in Sheets with all of the formatting from the original, though it may be necessary to perform the additional step of selecting ("Open in Sheets") once the user has uploaded the file to their drive.
  - SAFI_clean.csv also opens in Sheets with no issue (see above for the additional step if uploading to Drive).
- Lesson 3, Formatting Problems (no exercises)
- Lesson 4, Dates as Data
  - After opening dates.xlsx in Sheets, at least in my locale, the MONTH, DAY, and YEAR functions work as expected on the MM_DD_YEAR tab. On the DD_MM_YEAR tab, they produce a VALUE error, much like in Excel. I am not sure how Google Sheets would behave in locales where DD-MM-YEAR is the standard format.
  - The "default year" exercise also behaves the same in Sheets as in Excel.
- Lesson 5, Quality Assurance
  - It is not trivial to limit the user's input in the no_membrs column to whole numbers. This is possible but requires entering a custom formula into the Data Validation menu: `=IF(D124=INT(D124),True,False)`.
  - It is easy to restrict the user's input to a range of numbers, but non-integers are not automatically excluded.
  - Restricting entries to a list of hard-coded values, or to a list of values from another range, works as in Excel.
  - The functionality found in Excel under the Input Message and Error Alert tabs is present in the Sheets Validation popup, but it's more limited. (The user can set custom help text but cannot set a custom error message.)
- Lesson 6, Exporting Data
  - In Sheets, the user should select File--Download--Comma-separate values (as opposed to File--Save in Excel).

In summary, I think Sheets would probably be a viable option for those without access to Excel. (And although this topic doesn't come up in the Lessons, I have found in my own work that Sheets can be easier to work with than Excel when importing CSV data from other sources. Excel, for instance, tends to make assumptions about certain data types -- e.g., dates and long integers -- that can prove problematic if you don't correct for them on input. Sheets's type system seems a bit more flexible in this respect.)

(This comment submitted as part of the Carpentries Instructor Training Checkout process.)

**Comment by chris-prener** [comment]
thanks so much @dolsysmith - this is wonderful feedback! It looks like moving forward on a Google Sheets version of this is a viable option.

**Comment by bkmgit** [comment]
Made a pull request for a version using LibreOffice #121 . LibreOffice is also available as a cloud version through [Collabora Office](https://www.collaboraoffice.com/).  [Onlyoffice](https://www.onlyoffice.com/) is also available in the cloud, but the data validation section would need to be changed for this. [WPS Office](https://www.wps.com/) has an online version that at present is free to use and would allow for minimal additions to the validation section.

**Comment by lvermeyden** [comment]
If help is still need to update the lesson to work with google sheets I am happy to assist with that.

**Comment by kerchner** [comment]
Thinking about the learning objectives of this workshop, they transcend any particular spreadsheet program, so if someone later has data that requires a more secure environment, they will easily be able to apply what they learned in this lesson to a different program.   Google Sheets, equally accessible to everyone, removes the software setup barrier for Carpentries lesson participants, and could result in all participants being in the exact same program.  In my opinion that pedagogical benefit far outweighs any future concerns about working with sensitive data outside of the workshop - where, again, the lessons learned from the workshop could be applied in a different spreadsheet program.

**Comment by josenino95** [comment]
We recently talked with @allie-tatarian about this issue, and we though that now we shouldn't add Google Sheets in the lesson. I added some suggested text about it on issue #189. Feel free to review it and make comments.

You can find a rendered version of the changes made here:

[https://josenino95.github.io/spreadsheets-socialsci/instructor/index.html#software](https://josenino95.github.io/spreadsheets-socialsci/instructor/index.html#software)

**Comment by bkmgit** [comment]
https://ethercalc.net might be an alternative for browser based software.

**Comment by josenino95** [comment]
Thank you @bkmgit for mentioning that tool. Do you use EtherCal frequently or know if there's a substantial user base of that program? And another question: Would you suggest to include EtherCalc instead of other mentioned alternatives (Apple Numbers, Gnumeric, Onlyoffice, WPS office)?

I'm thinking the list of alternatives is long already, and wouldn't be in favor of adding another one. So if you know that EtherCal is a better suggestion, we can add it and remove a less used alternative.
